lib/spack/spack/build_systems/cmake.py

```diff
@@ -246,3 +256,80 @@ def check(self):
 
     # Check that self.prefix is there after installation
     run_after('install')(PackageBase.sanity_check_prefix)
+
+    def write_spconfig(self, spack_env, spconfig_fname, dirty):
+        fname = os.path.join(spack_env.path, spconfig_fname) \
+            if spack_env is not None else spconfig_fname
+
+        with open(fname, 'w') as fout:
+            self._write_spconfig(fout, dirty)
+            fout.write('\nproc = subprocess.Popen(cmd, env=env)\n'
+                       'proc.wait()\n')
+
+        set_executable(fname)
+        return fname
+
+    def _write_spconfig(self, fout, dirty):
+        """Writes the spconfig.py file to a stream."""
+
+        # Set-up the environment
+        _cmd = [str(which('cmake'))] + \
+            self.std_cmake_args + self.cmake_args()
+
+        # No verbose makefile for interactive builds
+        cmd = [x for x in _cmd if not x.startswith('-DCMAKE_VERBOSE_MAKEFILE')]
+
+        env = dict()
+
+        paths = os.environ['PATH'].split(':')
+        paths = [item for item in paths if 'spack/env' not in item]
+        env['PATH'] = ':'.join(paths)
+        env['SPACK_TRANSITIVE_INCLUDE_PATH'] = spack_transitive_include_path()
+        env['CMAKE_PREFIX_PATH'] = os.environ['CMAKE_PREFIX_PATH']
+
+        if 'SPACK_CC' in os.environ:
+            env['CC'] = os.environ['SPACK_CC']
+        if 'SPACK_CXX' in os.environ:
+            env['CXX'] = os.environ['SPACK_CXX']
+        if 'SPACK_FC' in os.environ:
+            env['FC'] = os.environ['SPACK_FC']
+
+        fout.write(
+            r"""#!%s
+#
+# %s
+
+import sys
+import os
+import subprocess
+
+def cmdlist(str):
+    return list(x.strip().replace("'",'') for x in str.split('\n') if x)
+env = dict(os.environ)
```

Comment:
@citibeth should this not be here? It is done conditionally right after this block

Reply:
See message on my recent commit.